### DISCUSSION TITLE
Way to get current xml in diagram in XML format using javascript

### DISCUSSION THREAD

**Discussion by reinoutkoninkx** [jgraph/drawio#4469] [Q&A]
Is there currently a way to retrieve the XML that is in the diagram? Using the postMessage API I would like to fire an action at the iframe that in turn would return an event that I can capture with the current XML in the diagram. Kinda the same as it is possible to capture the event for merge, autosave, etc..

Thanks in advance

**Comment by alderg** [comment]
Try `{action: 'export', format: 'xml'}`. See https://www.drawio.com/doc/faq/embed-mode
